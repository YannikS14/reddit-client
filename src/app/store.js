import { configureStore } from '@reduxjs/toolkit';
import postsReducer from '../features/posts/postsSlice';
import subredditReducer from '../features/subreddits/subredditsSlice';
import searchFieldReducer from '../features/searchField/searchFieldSlice';

export const store = configureStore({
  reducer: {
    posts: postsReducer,
    subreddits: subredditReducer,
    searchField: searchFieldReducer,
  },
});
