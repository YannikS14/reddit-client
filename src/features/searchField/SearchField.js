import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
  selectSearchText,
  updateSearchText,
} from './searchFieldSlice';

export default function SearchField() {
  const dispatch = useDispatch();
  const searchText = useSelector(selectSearchText);

  return (
    <div id="search" className="flex flex-1 mx-4 max-w-sm relative">
      <input
        type="search"
        placeholder="Search Reddit"
        value={searchText}
        onChange={(e) => dispatch(updateSearchText(e.target.value))}
        className="flex-1 border-solid border-2 border-gray-300 rounded-md pl-10 pr-4 py-2"
      ></input>
      <svg
        xmlns="http://www.w3.org/2000/svg"
        className="h-5 w-5 absolute left-4 top-3  text-gray-700"
        fill="none"
        viewBox="0 0 24 24"
        stroke="currentColor"
      >
        <path
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
        />
      </svg>
    </div>
  );
}
