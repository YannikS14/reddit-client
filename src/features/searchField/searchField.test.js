import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../app/store';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SearchField from './SearchField';

describe('SearchField component', () => {
  it('should update and display the value of the search input field', () => {
    render(
      <Provider store={store}>
        <SearchField />
      </Provider>,
    );
    expect(screen.getByRole('searchbox')).toHaveTextContent('');

    userEvent.type(screen.getByRole('searchbox'), 'Test Search');
    expect(screen.getByRole('searchbox')).toHaveValue('Test Search');

    userEvent.clear(screen.getByRole('searchbox'));
    expect(screen.getByRole('searchbox')).toHaveTextContent('');
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <SearchField />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
