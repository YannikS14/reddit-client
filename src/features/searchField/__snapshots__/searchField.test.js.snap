// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField component matches the snapshot 1`] = `
<div
  className="flex flex-1 mx-4 max-w-sm relative"
  id="search"
>
  <input
    className="flex-1 w-full border-solid border-2 dark:bg-gray-600 border-gray-300 dark:border-gray-500 rounded-md pl-10 pr-4 py-2"
    onChange={[Function]}
    placeholder="Search Reddit"
    type="search"
    value=""
  />
  <svg
    className="h-5 w-5 absolute left-4 top-3 text-gray-700 dark:text-gray-300"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
    />
  </svg>
</div>
`;
